src/planner/vectorfield/detail/BodyNodePoseVectorFieldConstraint.hpp

```diff
@@ -0,0 +1,44 @@
+#ifndef AIKIDO_PLANNER_VECTORFIELD_BODYNODEPOSEVECTORFIELDCONSTRAINT_HPP_
+#define AIKIDO_PLANNER_VECTORFIELD_BODYNODEPOSEVECTORFIELDCONSTRAINT_HPP_
+
+#include <aikido/constraint/Testable.hpp>
+#include <aikido/statespace/dart/MetaSkeletonStateSpace.hpp>
+
+namespace aikido {
+namespace planner {
+namespace vectorfield {
+namespace detail {
+
+/// This class implements a constraint for body node pose vector field.
+/// It checks a given state constraint and a bound constraint of state space.
+///
+class BodyNodePoseVectorFieldConstraint : public aikido::constraint::Testable
```

Comment:
I don't see why this class is necessary. It seems like we could achieve the same result by creating a `CompoundTestable` constraint that wraps `stateConstraint` and the joint limit constraint returned by calling `createTestableBounds()` on the `MetaSkeletonStateSpace`.

Reply:
I am not sure whether there is a standard on creating a `CompoundTestable`. I could not find any in `aikido`. But this class actually implements  a "compoundTestable" that wraps `stateConstraint` and the joint limit constraint returned by calling `createTestableBounds()` on the `MetaSkeletonStateSpace`.